examples/ExampleCableSpan.cpp

```
@@ -0,0 +1,252 @@
+
+        // Add ellipsoid obstacle.
+        cable.addObstacle(
+            matter.Ground(),
+            Transform(Vec3{-2., 0., 0.}),
+            std::shared_ptr<ContactGeometry>(
+                new ContactGeometry::Ellipsoid({1.5, 2.6, 1.})),
+            {0.0, 0., 1.1});
+
+        // Add sphere obstacle.
+        cable.addObstacle(
+            matter.Ground(),
+            Transform(Vec3{2., 0., 0.}),
+            std::shared_ptr<ContactGeometry>(new ContactGeometry::Sphere(1.)),
+            {0.1, 1.1, 0.}
+        );
+
+        // Add cylinder obstacle.
+        cable.addObstacle(
+            matter.Ground(),
+            Transform(Rotation(0.5*Pi, XAxis), Vec3{5., 0., 0.}),
+            std::shared_ptr<ContactGeometry>(new ContactGeometry::Cylinder(1.)),
+            Vec3{0., -1., 0.}
+        );
+
+        // Visaulize the system.
+        system.setUseUniformBackground(true); // no ground plane in display
+        Visualizer viz(system);
+        viz.addDecorationGenerator(new CableDecorator(system, cable));
+        ShowStuff showStuff(cables, 1e-3);
+
+        // Initialize the system and s.
+        system.realizeTopology();
+        State s = system.getDefaultState();
+        system.realize(s, Stage::Position);
+
+        system.realize(s, Stage::Report);
+        viz.report(s);
+        showStuff.handleEvent(s);
+
+        std::cout << "Hit ENTER ..., or q\n";
+        const char ch = getchar();
+        if (ch == 'Q' || ch == 'q') {
+            return 0;
+        }
+
+        Real angle = 0.;
+        while (true) {
+            system.realize(s, Stage::Position);
+            cable.calcLength(s);
+            viz.report(s);
+
+            // Move the cable origin.
+            angle += 0.01;
+            cableOriginBody.setQ(
+                s,
+                Vec3(sin(angle), 5. * sin(angle * 1.5), 5. * sin(angle * 2.)));
+        }
+
+    } catch (const std::exception& e) {
+        std::cout << "EXCEPTION: " << e.what() << "\n";
```

could probably just not catch it and let it bubble up?

---

It appears to be good practice, unifying unwinding of the stack over different platforms.

I also see this often (but not in the other examples):

```cpp
void doMain() { ...}

int main() {
    try {
        doMain();
    } catch (...) {...}
}
```

I'll leave it as is for consistency.